services/token_update_nfts.proto

```diff
@@ -34,19 +34,34 @@ import "google/protobuf/wrappers.proto";
 /**
  * At consensus, updates an already created Non Fungible Token to the given values.
  * 
- * If no value is given for a field, that field is left unchanged. For an immutable tokens (that is,
- * a token without an admin key), only the expiry may be updated. Setting any other field in that
- * case will cause the transaction status to resolve to TOKEN_IS_IMMUTABLE.
+ * If no value is given for a field, that field is left unchanged.
+ * Only certain fields such as metadata can be updated. For an immutable token (that is, a token
+ * without an admin key), only the expiry may be updated. Setting any other field in that case will
+ * cause the transaction status to resolve to TOKEN_IS_IMMUTABLE.
```

Comment:
is this true? Can we not change token metadata / nft metadata if the token has a metadata key but no admin key?

Reply:
We will be able to change metadata on the Nft if there is `metadata_key` on the token. Admin key is required only to update `metadata_key` on the token, similar to other keys on token.

Reply:
Then the comment is wrong and should be removed, i.e., the part starting from `For an immutable token`, believe it's copied from token_update.proto

Reply:
Agreed. Fixed @xin-hedera

Reply:
My understanding is that when updating metadata (token level or some number of NFTs by serial number), the transaction must be signed by the metadata key. With no metadata key defined, metadata is effectively immutable.

But, it sounds like the admin key can be used to later set a metadata key, making metadata mutable once again. The only way to guarantee metadata immutability is to first remove the metadata key, then the admin key. Just removing the admin key will leave metadata forever mutable since the metadata key is now immutable.

If I'm not just misunderstanding all this, then it seems we need to be clear on the order of operations required to reach a certain result. That is, if I bought a specific NFT because its metadata plays a part in defining its value and I expect that to be unchanging (not the dynamic stone sword to diamond sword scenario), the token may be technically immutable once the admin key is cleared, but if the metadata key still exists it seems there is a problem. In the sword upgrade scenario, then the metadata key is needed regardless of the admin key.

Reply:
HIP-540, which we are going to implement for the metadata key, too, describes this in detail. If there is no admin key, the metadata key can remove itself making the token immutable.